src/main/java/uk/gov/hmcts/reform/idam/service/UserRoleService.java

```diff
@@ -20,16 +21,26 @@ public class UserRoleService {
 
     private final RoleAssignmentClient roleAssignmentClient;
 
+    @Retry(retryAttempts = 2)
     public List<String> filterUsersWithRoles(List<String> staleUsers) {
+        if (staleUsers.isEmpty()) {
+            return List.of();
+        }
         var roleAssignmentQuery = UserRoleAssignmentQueryRequest.builder().userIds(staleUsers).build();
         UserRoleAssignmentQueryRequests body = UserRoleAssignmentQueryRequests.builder()
             .queryRequests(roleAssignmentQuery)
             .build();
 
-        final RoleAssignmentResponse response = roleAssignmentClient.getRoleAssignments(
-            Map.of("Content-Type", Constants.ROLE_ASSIGNMENTS_CONTENT_TYPE),
-            body
-        );
+        final RoleAssignmentResponse response;
+        try {
+            response = roleAssignmentClient.getRoleAssignments(
+                Map.of("Content-Type", Constants.ROLE_ASSIGNMENTS_CONTENT_TYPE),
+                body
+            );
+        } catch (Exception e) {
+            log.error("UserRoleService.getRoleAssignemnts threw exception: {}", e.getMessage(), e);
+            throw e;
```

Comment:
Can you please rethrow custom exception instead, UserRoleServiceException for example.